tests/index.test.ts

```diff
@@ -143,6 +143,11 @@ describe('index', () => {
       URIError,
       AggregateError,
     };
+    // Since the replacer will remove any undefined properties from the
+    // stringified result, we cannot expect that potentially undefined
+    // properties (such as the cause) will be present in a serialised error
+    // On the deserialisation side, undefined values will be recreated as
+    // undefined
```

Comment:
It's better for this comment to go into the docblock of `fromJSON` on `AbstractError`.

Reply:
Actually that should go into the `AbstractError.toJSON` method.

Reply:
So remove the comment from above the replacer completely and add
```
The `replacer` will:
* delete undefined values in objects
* replace undefined values for null in arrays 
```
to the docblock of `toJSON`?